### PR TITLE
Add monthly active users

### PR DESCRIPTION
**Pull request by pfefferle**
for better stats on FediDB

### CODE REVIEW

File: includes/functions.php [hunk 1]
```diff
@@ -474,3 +475,91 @@ function is_json( $data ) {
...
+ */
+function get_active_users( $duration = '1 month ago' ) {
+	// get all distinct authors that have published a post in the last 30 days
+	$posts = get_posts(
+		array(
+			'post_type'      => 'post',
+			'post_status'    => 'publish',
+			'orderby'        => 'post_count',
+			'order'          => 'DESC',
+			'posts_per_page' => 4,
```
Comment (mattwiebe) [inline includes/functions.php:494]: Having this in the query will cause problems, I think? 

The query will be limited to 4 posts, which means that only a maximum of 4 users would be able to be active on a given site, and also if, say, one prolific author has published the 4 most recent posts, only that author will be counted?

Comment (mattwiebe) [reply includes/functions.php:494]: I think we can probably do this better with a custom SQL query. We would probably then also want to cache the query. I asked ChatGPT for help and it looks pretty promising:

```php
global $wpdb;
$query = "
  SELECT COUNT(DISTINCT post_author) 
  FROM {$wpdb->posts} 
  WHERE post_type = 'post' 
  AND post_status = 'publish'
  AND post_date >= DATE_SUB(NOW(), INTERVAL 1 MONTH)
";
$num_authors = $wpdb->get_var($query);
```

We could wrap it all in a transient with a `DAY_IN_SECONDS` expiry?

**Review by mattwiebe — COMMENTED** [review]
Left some ideas

File: includes/functions.php (hunk 1 above)
Comment (pfefferle) [reply includes/functions.php:494]: Why have I added a posts per page? 😳🙄

**Comment by pfefferle** [comment]
@mattwiebe can you re-check?

**Review by mattwiebe — APPROVED** [review]
This looks good to me now :)

**Comment by pfefferle** [comment]
👍

But Why do you prefer transients?

**Comment by mattwiebe** [comment]
> But Why do you prefer transients?

WP mostly runs on shared hosts, which do not have an object cache, so the `wp_cache_*` functions in those cases do not persist beyond a single request. Transients were designed to overcome this limitation, by instead leveraging the Options API to store the results of slow things like network calls or expensive computations, like our custom query.

For hosts with an object cache (like WordPress.com with `memcached`, the transients API will use that instead of Options.

https://developer.wordpress.org/apis/transients/

**Pull request merged**